substrate/frame/session/src/lib.rs

```diff
@@ -436,8 +441,21 @@ pub mod pallet {
 
 		/// Weight information for extrinsics in this pallet.
 		type WeightInfo: WeightInfo;
+		
+		/// The currency type for reserving when setting keys.
+		type Currency: NamedReservableCurrency<Self::AccountId>;
+		
+		/// The reserve identifier type.
+		type ReserveIdentifier: Get<<Self::Currency as NamedReservableCurrency<Self::AccountId>>::ReserveIdentifier> + TypeInfo + 'static;
```

Comment:
We should directly use [`Holds`](https://paritytech.github.io/polkadot-sdk/master/frame_support/traits/tokens/fungibles/hold/trait.Mutate.html#method.hold)

Reply:
i have just pushed a fix to replace Reserve by Hold instead